spec/std/array_spec.cr

```diff
@@ -82,19 +94,45 @@ describe "Array" do
       b = [4, 5, 6] * 10
       (a | b).should eq([1, 2, 3, 4, 5, 6])
     end
-  end
 
-  it "does +" do
-    a = [1, 2, 3]
-    b = [4, 5]
-    c = a + b
-    c.size.should eq(5)
-    0.upto(4) { |i| c[i].should eq(i + 1) }
+    it "disjoint types" do
+      a = [1] | ['a']
+      a.should eq([1, 'a'])
+      typeof(a).should eq(Array(Int32 | Char))
+    end
+
+    it "union of arrays" do
+      a = [1, 'a'] | ([1, true] || [1, ""])
+      a.should eq([1, 'a', true])
+      typeof(a).should eq(Array(Int32 | Char | Bool | String))
+    end
   end
 
-  it "does + with empty tuple converted to array (#909)" do
-    ([1, 2] + Tuple.new.to_a).should eq([1, 2])
-    (Tuple.new.to_a + [1, 2]).should eq([1, 2])
+  describe "+" do
+    it "same element type" do
+      a = [1, 2, 3]
+      b = [4, 5]
+      c = a + b
+      c.size.should eq(5)
+      0.upto(4) { |i| c[i].should eq(i + 1) }
```

Comment:
Writing the expected result as an explicit literal is concise than constructing it programmatically.

```suggestion
      c.should eq [1, 2, 3, 4, 5]
```